binding/client.go

```diff
@@ -422,9 +422,19 @@ func (r *Client) FileGet(path, destination string) (err error) {
 }
 
 //
-// FilePut uploads a file.
+// FilePost uploads a file.
 // Returns the created File resource.
-func (r *Client) FilePut(path, source string, object interface{}) (err error) {
+func (r *Client) FilePost(path, source string, object interface{}) (err error) {
+	if source == "" {
+		fields := []Field{
+			{
+				Name:   api.FileField,
+				Reader: bytes.NewReader([]byte{}),
+			},
+		}
+		err = r.FileSend(path, http.MethodPut, fields, object)
```

Comment:
Looks like this should be `http.MethodPost`

Reply:
Fixed.